app/backend/src/couchers/models.py

```diff
@@ -1068,3 +1068,65 @@ class Reply(Base):
     deleted = Column(DateTime(timezone=True), nullable=True)
 
     comment = relationship("Comment", backref="replies")
+
+
+class BackgroundJobType(enum.Enum):
+    # payload: jobs.SendEmailPayload
+    send_email = 1
+    # payload: google.protobuf.Empty
+    purge_login_tokens = 2
+    # payload: google.protobuf.Empty
+    purge_signup_tokens = 3
```

Comment:
Remove repeated jobs from database enum.

Reply:
Could they not still be run through the background queue?

Reply:
You can still run the jobs every 24 hours, just don't add them into in the database schema please, since they already store the state they need in the other tables.

Reply:
The abstraction here is that a background job is some longer-running or async operation that should be retried. It takes a certain protobuf as an input and produces no output. A failing job is retried a few times with exponential backoff. Background jobs are serviced by a set of workers asynchronously from a job queue. The tasks themselves are defined in the same file and the job type to function lookup is centralized.

Some jobs are run as a response to user input, e.g. sending notifications or links.

Some jobs are additionally run according to some rough schedule, but may also be triggered manually.

I'd like this scheme. I'd like the repeated jobs to also take advantage of the other features of background jobs.

Also I'd really not feel like rewriting all the tests as I used the two background jobs as generic jobs (since they're idempotent and shouldn't fail) and so most tests utilize those (I need at least two jobs for testing).

I understand this model has its downsides, and depending on how the world unfolds, we probably need to add functionality and modify it as we go. But my opinion is that it's pretty good for now.

I don't see what we get from separating the repeating jobs from the non-repeating jobs, except a small decrease in complexity. But then we have two job systems: repeating and non-repeating. It won't make sense (and will be confusing) to have all the tasks in the same file if they don't behave the same, so they should be split up.

Reply:
I disagree. I my world each job keep tracks of its own state independent of other jobs. The interval each job is run at is stored in the Python source code and not in the database. Also in my world, token purging never fails :)

But as you say we could always change it later. I'll make a final codewise review of the PR!

Reply:
I will merge this, but let's have a brief chat again in the backend meeting :)